streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -513,22 +507,31 @@ class StreamingContext private[streaming] (
   }
 
   /**
+   * Return the current state of the context.
+   */
+  def getState(): StreamingContextState = {
+    state
+  }
+
+  /**
    * Start the execution of the streams.
    *
    * @throws SparkException if the context has already been started or stopped.
    */
   def start(): Unit = synchronized {
-    if (state == Started) {
-      throw new SparkException("StreamingContext has already been started")
-    }
-    if (state == Stopped) {
-      throw new SparkException("StreamingContext has already been stopped")
+    state match {
+      case INITIALIZED =>
+        // good to start
+      case STARTED =>
```

Comment:
I agree with you. In this PR I preserved existing PR. I changed it to a no-op in #6060